diff --git a/queries/highlights.scm b/queries/highlights.scm
index b13b4f460fa80860b552070d7d9d2f527b164381..d1d01430e592bef630ff0c1d55dde5bef686fabe 100644
--- a/queries/highlights.scm
+++ b/queries/highlights.scm
@@ -147,3 +147,23 @@
   "with"
   "yield"
 ] @keyword
+
++
+(type_arguments
+  "<" @punctuation.bracket
+  ">" @punctuation.bracket)
+
+[
+  "("
+  ")"
+  "["
+  "]"
+  "{"
+  "}"
+] @punctuation.bracket
+
+[
+  "."
+  ","
+  ";"
+] @punctuation.delimiter
