diff --git a/queries/highlights.scm b/queries/highlights.scm
index 9312d68288070c3c2b90eb8275e8031661c95850..e03257cccc2d2f4a7fe49d187a59f67924038dc4 100644
--- a/queries/highlights.scm
+++ b/queries/highlights.scm
@@ -1,15 +1,23 @@
 ; Variables
-;----------
 
 (identifier) @variable
 
 ; Properties
-;-----------
 
 (property_identifier) @property
+(shorthand_property_identifier) @property
+(shorthand_property_identifier_pattern) @property
+
+; Function and method calls
+
+(call_expression
+  function: (identifier) @function)
+
+(call_expression
+  function: (member_expression
+    property: (property_identifier) @function.method))
 
 ; Function and method definitions
-;--------------------------------
 
 (function_expression
   name: (identifier) @function)
@@ -35,50 +43,33 @@
   left: (identifier) @function
   right: [(function_expression) (arrow_function)])
 
-; Function and method calls
-;--------------------------
-
-(call_expression
-  function: (identifier) @function)
-
-(call_expression
-  function: (member_expression
-    property: (property_identifier) @function.method))
-
 ; Special identifiers
-;--------------------
 
-((identifier) @constructor
- (#match? @constructor "^[A-Z]"))
+((identifier) @type
+ (#match? @type "^[A-Z]"))
 
 ([
-    (identifier)
-    (shorthand_property_identifier)
-    (shorthand_property_identifier_pattern)
+  (identifier)
+  (shorthand_property_identifier)
+  (shorthand_property_identifier_pattern)
  ] @constant
- (#match? @constant "^[A-Z_][A-Z\\d_]+$"))
-
-((identifier) @variable.builtin
- (#match? @variable.builtin "^(arguments|module|console|window|document)$")
- (#is-not? local))
-
-((identifier) @function.builtin
- (#eq? @function.builtin "require")
- (#is-not? local))
+ (#match? @constant "^_*[A-Z_][A-Z\\d_]*$"))
 
 ; Literals
-;---------
 
-(this) @variable.builtin
-(super) @variable.builtin
+(this) @variable.special
+(super) @variable.special
 
 [
-  (true)
-  (false)
   (null)
   (undefined)
 ] @constant.builtin
 
+[
+  (true)
+  (false)
+] @boolean
+
 (comment) @comment
 
 [
@@ -86,17 +77,18 @@
   (template_string)
 ] @string
 
-(regex) @string.special
+(escape_sequence) @string.escape
+
+(regex) @string.regex
 (number) @number
 
 ; Tokens
-;-------
 
 [
   ";"
-  (optional_chain)
   "."
   ","
+  ":"
 ] @punctuation.delimiter
 
 [
@@ -155,9 +147,12 @@
   "}"
 ]  @punctuation.bracket
 
-(template_substitution
-  "${" @punctuation.special
-  "}" @punctuation.special) @embedded
+(ternary_expression
+  [
+    "?"
+    ":"
+  ] @operator
+)
 
 [
   "as"
@@ -202,3 +197,7 @@
   "with"
   "yield"
 ] @keyword
+
+(template_substitution
+  "${" @punctuation.special
+  "}" @punctuation.special) @embedded
