diff --git a/node_modules/tree-sitter-go/.bun-tag-160cc8962402766d b/.bun-tag-160cc8962402766d
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/tree-sitter-go/.bun-tag-86b1ea0e53ea9ae8 b/.bun-tag-86b1ea0e53ea9ae8
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/queries/highlights.scm b/queries/highlights.scm
index 6a3c0ac8b821db27582a2f378385a0a1c3a91b0d..e3881ad7bf0e5feafc71127c5b8e4fad681c064e 100644
--- a/queries/highlights.scm
+++ b/queries/highlights.scm
@@ -1,31 +1,32 @@
-; Function calls
+(type_identifier) @type
+(field_identifier) @variable.member
 
-(call_expression
-  function: (identifier) @function)
+(keyed_element
+  .
+  (literal_element
+    (identifier) @variable.member))
 
 (call_expression
-  function: (identifier) @function.builtin
-  (#match? @function.builtin "^(append|cap|close|complex|copy|delete|imag|len|make|new|panic|print|println|real|recover)$"))
+  function: (identifier) @function)
 
 (call_expression
   function: (selector_expression
     field: (field_identifier) @function.method))
 
-; Function definitions
-
 (function_declaration
   name: (identifier) @function)
 
 (method_declaration
   name: (field_identifier) @function.method)
 
-; Identifiers
-
-(type_identifier) @type
-(field_identifier) @property
-(identifier) @variable
-
-; Operators
+[
+  "("
+  ")"
+  "{"
+  "}"
+  "["
+  "]"
+] @punctuation.bracket
 
 [
   "--"
@@ -65,10 +66,9 @@
   "|="
   "||"
   "~"
+  ":"
 ] @operator
 
-; Keywords
-
 [
   "break"
   "case"
@@ -97,15 +97,13 @@
   "var"
 ] @keyword
 
-; Literals
-
 [
   (interpreted_string_literal)
   (raw_string_literal)
   (rune_literal)
 ] @string
 
-(escape_sequence) @escape
+(escape_sequence) @string.escape
 
 [
   (int_literal)
@@ -113,6 +111,9 @@
   (imaginary_literal)
 ] @number
 
+(const_spec
+  name: (identifier) @constant)
+
 [
   (true)
   (false)
@@ -121,3 +122,9 @@
 ] @constant.builtin
 
 (comment) @comment
+
+[
+  "."
+  ","
+  ";"
+] @punctuation.delimiter
