diff --git a/queries/highlights.scm b/queries/highlights.scm
index 763661af7616fe25894d019247e83032ac4e2526..4ddfe9a418ca700c57c135e01a53285266d93321 100644
--- a/queries/highlights.scm
+++ b/queries/highlights.scm
@@ -1,64 +1,92 @@
 (comment) @comment
 
-(tag_name) @tag
-(nesting_selector) @tag
-(universal_selector) @tag
+[
+  (tag_name)
+  (nesting_selector)
+  (universal_selector)
+] @tag
 
-"~" @operator
-">" @operator
-"+" @operator
-"-" @operator
-"*" @operator
-"/" @operator
-"=" @operator
-"^=" @operator
-"|=" @operator
-"~=" @operator
-"$=" @operator
-"*=" @operator
-
-"and" @operator
-"or" @operator
-"not" @operator
-"only" @operator
+[
+  "~"
+  ">"
+  "+"
+  "-"
+  "*"
+  "/"
+  "="
+  "^="
+  "|="
+  "~="
+  "$="
+  "*="
+  "and"
+  "or"
+  "not"
+  "only"
+] @operator
 
 (attribute_selector (plain_value) @string)
+
+(attribute_name) @attribute
 (pseudo_element_selector (tag_name) @attribute)
 (pseudo_class_selector (class_name) @attribute)
 
-(class_name) @property
-(id_name) @property
-(namespace_name) @property
-(property_name) @property
-(feature_name) @property
+[
+  (class_name)
+  (id_name)
+  (namespace_name)
+  (feature_name)
+] @property
 
-(attribute_name) @attribute
+(property_name) @constant
 
 (function_name) @function
 
-((property_name) @variable
- (#match? @variable "^--"))
-((plain_value) @variable
- (#match? @variable "^--"))
+(
+  [
+    (property_name)
+    (plain_value)
+  ] @variable.special
+  (#match? @variable.special "^--")
+)
 
-"@media" @keyword
-"@import" @keyword
-"@charset" @keyword
-"@namespace" @keyword
-"@supports" @keyword
-"@keyframes" @keyword
-(at_keyword) @keyword
-(to) @keyword
-(from) @keyword
-(important) @keyword
+[
+  "@media"
+  "@import"
+  "@charset"
+  "@namespace"
+  "@supports"
+  "@keyframes"
+  (at_keyword)
+  (to)
+  (from)
+  (important)
+]  @keyword
 
 (string_value) @string
 (color_value) @string.special
 
-(integer_value) @number
-(float_value) @number
+[
+  (integer_value)
+  (float_value)
+] @number
+
 (unit) @type
 
-"#" @punctuation.delimiter
-"," @punctuation.delimiter
-":" @punctuation.delimiter
+[
+  ","
+  ":"
+  "."
+  "::"
+  ";"
+  "#"
+] @punctuation.delimiter
+
+[
+  "{"
+  ")"
+  "("
+  "}"
+  "["
+  "]"
+] @punctuation.bracket
