diff --git a/queries/highlights.scm b/queries/highlights.scm
index f33a7c2d3ae8c63364845dbd8cef1242ffa07f04..5ae8bcf4bc8f0b497dbcde4a1d7cc15c176bb7bc 100644
--- a/queries/highlights.scm
+++ b/queries/highlights.scm
@@ -3,6 +3,7 @@
   (raw_string)
   (heredoc_body)
   (heredoc_start)
+  (ansi_c_string)
 ] @string
 
 (command_name) @function
@@ -27,6 +28,8 @@
   "unset"
   "until"
   "while"
+  "local"
+  "declare"
 ] @keyword
 
 (comment) @comment
@@ -48,6 +51,7 @@
   ">>"
   "<"
   "|"
+  "="
 ] @operator
 
 (
